database/factories/ModelFactory.php

```diff
@@ -48,55 +61,67 @@
             ])->id;
         },
         'northstar_id' => $this->faker->northstar_id,
-        'url' => $url,
         'text' => $faker->sentence(),
         'location' => 'US-'.$faker->stateAbbr(),
-        // @TODO: Only set school if the action is set to collect school ID.
-        'school_id' => $this->faker->school_id,
+        /**
+         * Although our action models are set to collect school, not all users will create posts
+         * on behalf of their school (because they don't have a school_id set on their
+         * Northstar profile).
+         */
+        'school_id' => rand(0, 1) ? $this->faker->school_id : null,
         'source' => 'phpunit',
-        'status' => 'pending',
-        'quantity' => $faker->randomNumber(2),
     ];
 });
 
-// @TODO: These should all extend a photo-less "base" post, instead.
-$factory->defineAs(Post::class, 'text', function (Generator $faker) {
-    $faker->addProvider(new FakerNorthstarId($faker));
-    $faker->addProvider(new FakerSchoolId($faker));
+/**
+ * Photo post factory states.
+ */
+$factory->defineAs(Post::class, 'photo-accepted', function (Generator $faker) use ($factory) {
+    return array_merge($factory->raw(Post::class), [
```

Comment:
OMG look at this use of `array_merge`, this makes this SOOOOO much more readable!!! 🎉